packages/dashboard/src/pages/Personas/hooks/useContacts.ts

```diff
@@ -0,0 +1,43 @@
+import { useAsyncRetry, useUpdateEffect } from 'react-use'
+import { Services } from '../../../API'
+import type { Relation } from '@masknet/shared'
+import { useRef } from 'react'
+import { last } from 'lodash-es'
+
+export const useContacts = (network: string, page: number, size = 20) => {
+    const cache = useRef<Map<number, Relation[]>>(new Map([]))
```

Comment:
I suggest do not add a cache here. It's fragile and easy to get things wrong. We'll find a robust cache mechanism in the future.

Reply:
But I will not be able to complete the paging without a cache the Record of the last value on each page